python/packages/autogen-ext/src/autogen_ext/agents/openai/_openai_agent.py

```diff
@@ -0,0 +1,715 @@
+
+    content: str
+    call_id: str
+    name: str
+    is_error: bool = False
+
+
+class OpenAIAgent(BaseChatAgent, Component[OpenAIAgentConfig]):
+    """
+    An agent implementation that uses the OpenAI Responses API to generate responses.
+
+    Installation:
+
+    .. code-block:: bash
+
+        pip install "autogen-ext[openai]"
+        # pip install "autogen-ext[openai,azure]"  # For Azure OpenAI Assistant
+
+    This agent leverages the Responses API to generate responses with capabilities like:
+
+    * Custom function calling
+    * Multi-turn conversations
+
+    Example:
+
+        .. code-block:: python
+
+            from openai import AsyncOpenAI
+            from autogen_core import CancellationToken
+            from autogen_ext.agents.openai import OpenAIAgent
+            from autogen_agentchat.messages import TextMessage
+
+
+            async def example():
+                cancellation_token = CancellationToken()
+                client = AsyncOpenAI()
+                agent = OpenAIAgent(
+                    name="Simple Agent",
+                    description="A simple OpenAI agent using the Responses API",
+                    client=client,
+                    model="gpt-4.1",
+                    instructions="You are a helpful assistant.",
+                )
+                response = await agent.on_messages([TextMessage(source="user", content="Hello!")], cancellation_token)
+                print(response)
+
+        asyncio.run(example())
+
+
+    TODO: Add support for advanced features (vector store, multimodal, etc.) in future PRs.
+
+    """
+
+    component_config_schema = OpenAIAgentConfig
+    component_provider_override = "autogen_ext.agents.openai.OpenAIAgent"
+
+    def __init__(
+        self: "OpenAIAgent",
+        name: str,
+        description: str,
+        client: Union[AsyncOpenAI, AsyncAzureOpenAI],
+        model: str,
+        instructions: str,
+        tools: Optional[List[Tool]] = None,
+        temperature: Optional[float] = 1,
+        max_output_tokens: Optional[int] = None,
+        json_mode: bool = False,
+        store: bool = True,
+        truncation: str = "disabled",
+    ) -> None:
+        super().__init__(name, description)
+        self._client: Union[AsyncOpenAI, AsyncAzureOpenAI] = client
+        self._model: str = model
+        self._instructions: str = instructions
+        self._temperature: Optional[float] = temperature
+        self._max_output_tokens: Optional[int] = max_output_tokens
+        self._json_mode: bool = json_mode
+        self._store: bool = store
+        self._truncation: str = truncation
+        self._last_response_id: Optional[str] = None
+        self._message_history: List[Dict[str, Any]] = []
+        self._tools: List[Dict[str, Any]] = []
+        self._tool_map: Dict[str, Tool] = {}
+        if tools is not None:
+            for tool in tools:
+                function_schema: Dict[str, Any] = {
```

Comment:
On default, the Responses API provides lots of tool types such as file search, web search, computer use, mcp tool, code interpreter, image generation tool and local shell tool as well as function. Do you want to limit AutoGen OpenAIAgent's tool to just function?

Reply:
We need these features!

Reply:
But this clearly wasn't tested as Function tools aren't working (#6935 & #6937)

Reply:
@jay-thakur please see #6935 for the logs